### ISSUE TITLE
AdvancedDataGridColumn mx:itemEditor mx:Component should be Editable

### ISSUE THREAD

**Issue by pashminakazi** [apache/royale-asjs#731]
Project Details are :
https://paste.apache.org/cpo8y
Test case is available on VDI.
Go to Accounts Payable from Select in Menu
Go to ATM Rental Income from Setups in Menu
File : com/dbz/modules/AR/dbzATMRentalReceipt.mxml

Open this Setup Click on New button then on plus button (+)
Detail grid will be open then click plus button (+)  of detail grid,row will be added when you click on + button,that row should be editable like flex.
Royale : Nothing is editable,and on Clicking it is not opening Picklist
![image](https://user-images.githubusercontent.com/42200979/75154761-4ec80700-5730-11ea-8c6b-3a7bf16188e3.png)

Flex: When click on **Monthly Amount** Column,it is opening numericStepper
![image](https://user-images.githubusercontent.com/42200979/75154603-e416cb80-572f-11ea-84fa-4b9140e9fb9a.png)

When Click on 5th Column **Rental Income** it is opening Picklist in Flex
![image](https://user-images.githubusercontent.com/42200979/75154668-0f99b600-5730-11ea-898c-e1c1a256e9e3.png)

**Comment by aharui** [comment]
Try with latest changes and see if it works better

**Comment by pashminakazi** [comment]
I have downloaded new sdk today after your this commit https://github.com/apache/royale-asjs/commit/36cbf6ae96d642ad7839b564f003c2445562a320
And this is still not working

**Comment by aharui** [comment]
is the VDI up to date with these changes?

**Comment by pashminakazi** [comment]
Yes,vdi is git repo with latest sdk till your commit
https://github.com/apache/royale-asjs/commit/36cbf6ae96d642ad7839b564f003c2445562a320
And all setups related to issues are compiled with new sdk on vdi

**Comment by aharui** [comment]
I've pushed changes to get MXAdvancedDataGridItemRenderer to work, and to contain a DateField.  Some of the custom renderers in your app seem unusual:  An HBox wraps an MXAdvancedDataGridItemRenderer.  Are you sure your Flex version does that?  It doesn't seem like the HBox wrapper would allow proper handling of editing.

**Comment by pashminakazi** [comment]
On Clicking `Rental Income a/c `it is opening Picklist in Flex and In Royale it is not opening PickList
![image](https://user-images.githubusercontent.com/42200979/75656999-102dd180-5c87-11ea-87e2-055d43c1d42e.png)

**Comment by pashminakazi** [comment]
after this change Detail ADG is blank for Every row,before this commit ADG was full of rows
![image](https://user-images.githubusercontent.com/42200979/75657269-9cd88f80-5c87-11ea-8cfc-544169d2ce51.png)

**Comment by pashminakazi** [comment]
Yes, HBox wrapping MXAdvancedDataGridItemRenderer In Flex as well

**Comment by pashminakazi** [comment]
After this commit Exception On Opening ATM Rental Income Setup : 
![image](https://user-images.githubusercontent.com/42200979/75658454-2ee19780-5c8a-11ea-806e-87c092039394.png)
No other Exception,but now Detail ADG is blank

**Comment by aharui** [comment]
I saw an HBox wrapping an MXAdvancedDataGridItemRenderer containing a DateField.  The editorDataField for the column was referencing a property defined in the MXAdvancedDGIR, not the HBox.  I do not understand how that can work.  Please explain.  What is the purpose of the HBox?  MXAdvancedDGIR is normally the top-level tag.

**Comment by pashminakazi** [comment]
Sorry but i don't know the purpose of the HBox,I never worked on Flex

**Comment by aharui** [comment]
It really isn't my responsibility to find out why your Flex code used an HBox.  You are supposed to be more knowledgable about your code than me.  Ask your manager or the people who wrote the code or figure it out yourself by taking away the HBox.

In my test case, MXAdvancedDGIR as the top tag in the renderer worked.  I did not try HBox as the code did not make sense.

**Comment by aharui** [comment]
I took a look at the app as you left it on the VDI.  The detail ADG does not have any items in the DataProvider so it is correctly blank.  Lots of exceptions are being thrown by the "sample" DG because it has been shrunk down.  Since it has visible=false, you could give it a larger size and see if that makes the exception go away.  There is a chance that the exceptions are affecting initialization of the detail ADG

**Comment by pashminakazi** [comment]
After your latest commits the dataprovider is setting no values,but if I
revert the change the dataprovider sets the value in Detail grid

**Comment by aharui** [comment]
Try again with the latest code in case one of those fixes helped here.

**Comment by pashminakazi** [comment]
@aharui Picklist is not opening on Clicking 5th Column Rental Income and numericStepper values are not changing by clicking on top and bottom arrow of numericStepper,values change by typing numericStepper value

**Comment by aharui** [comment]
I do not see an editor specified for Rental Income column.

**Comment by pashminakazi** [comment]
In dbzATMRentalReceiptDetail.mxml
				`<mx:AdvancedDataGridColumn headerText="Rental Income a/c" dataField="rrd_Acc_Code" id="rrd_Acc_Code" width="120" editable="false" labelFunction="fn_FormatCOAGrid"/>`

```
	var arrPLTitle:Array = new Array( "Rental Income a/c", "Cost Center (Retail Outlet)");
					
					var arrPLCode:Array = new Array(arrDtlTransDP[j].rrd_Acc_Code, arrDtlTransDP[j].rrd_Costcntr_Code);
					
					var arrPLDesc:Array = new Array(arrDtlTransDP[j].rrd_Acc_Desc, arrDtlTransDP[j].rrd_Costcntr_Desc);
```
This code is setting value on Rental Income
and itemDoubleClick="fn_EditFieldDetail(event);" is applied on AdvancedDataGrid whenever any Record of ADG will be doubleClicked this function Called and **else if(colName == "rrd_Acc_Code")**  should be executed because **Rental Income a/c Column Id="rrd_Acc_Code"** and then it is opening Picklist in Code ```
parentMxmlDocument.parentMxmlDocument.parentMxmlDocument.f_HandlePickLists_Doc(dbzDataStructure.PL_COST_CENTER,
						dbzDataStructure.PL_MODE_SIMPLE, Costcntr_Code, Costcntr_Desc, null);
```	
```
public function fn_EditFieldDetail(event:ListEvent):void 
			{
				gi_row = event.rowIndex;
				gi_col = event.columnIndex
				var colName:String = dgTrans.columns[event.columnIndex].dataField;
				
				if(colName == "rrd_Costcntr_Code")
				{		
					Costcntr_Code.text = arrDtlTransDP[gi_row].rrd_Costcntr_Code;
					Costcntr_Desc.text = arrDtlTransDP[gi_row].rrd_Costcntr_Desc; 
					parentMxmlDocument.parentMxmlDocument.parentMxmlDocument.f_HandlePickLists_Doc(dbzDataStructure.PL_COST_CENTER,
						dbzDataStructure.PL_MODE_SIMPLE, Costcntr_Code, Costcntr_Desc, null);	//strDocTypeRef									
				}
				else if(colName == "rrd_Acc_Code")
				{			
					Acc_Code.text = arrDtlTransDP[gi_row].rrd_Acc_Code;
					Acc_Desc.text = arrDtlTransDP[gi_row].rrd_Acc_Desc; 
					parentMxmlDocument.parentMxmlDocument.parentMxmlDocument.f_HandlePickLists_Doc(dbzDataStructure.PL_ACCOUNT,
						dbzDataStructure.PL_MODE_SIMPLE, Acc_Code, Acc_Desc, null);	//strDocTypeRef									
				}				
			}
```

Please solve this issue

**Comment by pashminakazi** [comment]
@aharui Check this issue

**Comment by aharui** [comment]
I've made some changes that might fix the issues.

**Comment by pashminakazi** [comment]
@aharui Thank u so much it is working fine now,but one more thing
In detail Grid Monthly Amount is numericStepper,so the value of numericStepper should be changed by clicking up or down arrow of NumericStepper,but the value isn't change
![image](https://user-images.githubusercontent.com/42200979/76388576-422beb80-638b-11ea-9e31-1c3ca29b29b7.png)

**Comment by aharui** [comment]
That image implies you do not have all of the latest changes.  Are you saying when you click on the button nothing happens, or the NS goes away, or when the NS goes away that the value is not updated?
